### PR TITLE
BUG: Fix rfft for even input length.

### LINKED ISSUE #26349: BUG: behavioral change of numpy.fft.rfft with numpy 2.x
### Describe the issue:

With numpy 2.x:
```
% python -c 'import numpy as np; print(np.version.version); x = np.arange(8.) ; print(np.fft.rfft(x, 4))' 
2.1.0.dev0+git20240423.ef5f10d
[ 6.+0.j -2.+2.j -2.+4.j]
```

with numpy 1.26.4:

```
% python -c 'import numpy as np; print(np.version.version); x = np.arange(8.) ; print(np.fft.rfft(x, 4))'
1.26.4
[ 6.+0.j -2.+2.j -2.+0.j]
```

that's a notable output difference, maybe introduced by the vectorized form of pocketfft?

### Reproduce the code example:

```python
import numpy as np
x = np.arange(8.)
print(np.fft.rfft(x, 4))
```

### Error message:

_No response_

### Python and NumPy Versions:

2.1.0.dev0+git20240423.ef5f10d
3.12.2 (main, Feb 21 2024, 00:00:00) [GCC 13.2.1 20231205 (Red Hat 13.2.1-6)]

### Runtime Environment:

_No response_

### Context for the issue:

Difference found while porting pythran to numpy 2.x, as pythran mimics the 'legacy' behavior.
I'm unsure whether the new behavior is intended or not, at least it's not documented in the git log as far as I can tell (?)

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
Closes gh-26349

### CODE REVIEW

**Comment by ngoldbaum** [comment]
mac failure is an unrelated heisenbug

File: numpy/fft/tests/test_pocketfft.py [hunk 1]
```diff
@@ -258,6 +257,16 @@ def test_rfft(self):
                 np.fft.rfft(x, n=n) / n,
                 np.fft.rfft(x, n=n, norm="forward"), atol=1e-6)
 
+    def test_rfft_even(self):
+        x = np.arange(8)
+        y = np.fft.rfft(x, 4)
+        assert_allclose(y, [6.0, -2.0+2.0j, -2.0], rtol=1e-14)
```
Comment (charris) [inline numpy/fft/tests/test_pocketfft.py:263]: Could use the same sort of test as for the odd case, I like comparing to the full complex fft case.

Comment (WarrenWeckesser) [reply numpy/fft/tests/test_pocketfft.py]: Done.

**Comment by charris** [comment]
Thanks Warren.

File: numpy/fft/_pocketfft_umath.cpp [hunk 2]
```diff
@@ -183,12 +183,15 @@ rfft_impl(char **args, npy_intp const *dimensions, npy_intp const *steps,
          * Pocketfft uses FFTpack order, R0,R1,I1,...Rn-1,In-1,Rn[,In] (last
          * for npts odd only). To make unpacking easy, we place the real data
          * offset by one in the buffer, so that we just have to move R0 and
-         * create I0=0. Note that copy_data will zero the In component for
+         * create I0=0. Note that copy_input will zero the In component for
```
Comment (mhvk) [inline numpy/fft/_pocketfft_umath.cpp:186]: This is weird - so, the statement I made in the comment is not true? I'll try to have a quick look, but in any case the fix clearly works.

**Comment by mhvk** [comment]
#26359 does a follow-up that ensures the code actually does what the comment states. Would be nice to have both this one and #26359 be in 2.0.0....

Must admit that overall rewriting the fft routines in terms of ufuncs has been more of an experience in how hard it is to write bug-free code than I would have liked... Sorry all!

**Pull request merged**